packages/twenty-front/src/modules/object-record/record-merge/components/MergePreviewEffect.tsx

```diff
@@ -0,0 +1,54 @@
+import {
+  MergeManySettings,
+  useMergeManyRecords,
+} from '@/object-record/hooks/useMergeManyRecords';
+import { useUpsertRecordsInStore } from '@/object-record/record-store/hooks/useUpsertRecordsInStore';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { useEffect } from 'react';
+
+type MergePreviewEffectProps = {
+  objectNameSingular: string;
+  selectedRecords: ObjectRecord[];
+  mergeSettings: MergeManySettings;
+  onPreviewChange: (record: ObjectRecord | null) => void;
+  onLoadingChange: (isLoading: boolean) => void;
+};
+
+export const MergePreviewEffect = ({
+  objectNameSingular,
+  selectedRecords,
+  mergeSettings,
+  onPreviewChange,
+  onLoadingChange,
+}: MergePreviewEffectProps) => {
+  const { mergeManyRecords } = useMergeManyRecords({
+    objectNameSingular,
+  });
+  const { upsertRecords } = useUpsertRecordsInStore();
+
+  useEffect(() => {
```

Comment:
can we avoid this effect? Why do we need it here? I would expect merge action to be synchronously triggered

Reply:
actually got it, it's a separate tab. I don't think we can avoid it for now